src/DependencyInjection/Configuration.php

```diff
@@ -214,4 +215,23 @@ private function createSmsConfiguration(ArrayNodeDefinition $root)
                 ->end()
             ->end();
     }
+
+    private function createLocaleCookieConfiguration(ArrayNodeDefinition $root)
+    {
+        $root
+            ->children()
+                ->arrayNode('locale_cookie')
+                    ->info('Cookie settings for locale cookie')
+                    ->isRequired()
+                    ->children()
+                        ->scalarNode('name')->end()
+                        ->scalarNode('domain')->end()
+                        ->integerNode('expire')->end()
```

Comment:
Aah noo, that's not a pointer I was trying to give :stuck_out_tongue: The state `CookieSettings` had worked great I think! I feel it's kind of weird to pass around a `Cookie` with a null value to serve as a template for new `Cookie`s.